src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -493,4 +511,56 @@ public static Object coerceBodyType(InputStream body, ResponseBodyType as,
 
         return response;
     }
+
+    private static Timer.Context startTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
+                    requestLine.getMethod());
+            return registry.timer(name).time();
+        } else {
+            return null;
+        }
+    }
+
+    private static Timer.Context startUnbufferedStreamTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
+                    requestLine.getMethod(), "unbuffered_stream");
+            return registry.timer(name).time();
+        } else {
+            return null;
+        }
+    }
+
+    public static SortedMap<String, Timer> getClientMetrics(MetricRegistry metricRegistry){
```

Comment:
This namespace is under "impl" and is definitely a jumble of public and private methods, none of which have any javadocs (the javadocs are typically on the methods in the `(As|Sync)HttpClient` interfaces). I guess I can add javadocs to these new public methods if that's what we'd eventually like for all of the methods in this namespace?